[brendanng7] iP

brendanng7 opened the pull request:
# Soju
> My favourite emoji is 😭 and 🍻  makes me 😆 

## Features
- ability to store tasks in a list
- find tasks based on description

## Types of tasks
1. Todos
2. Deadlines
3. Events

The entry point to this project is in `./Soju`
```
public class Soju {
    public static void main(String[] args) {
        Soju soju = new Soju();
        soju.run();
    }
}
```

Some features I would like to add are:
- [ ] a drink command to track how much water you are drinking
- [ ] a way to assign tasks to names
- [x] better error handling

_Download [Soju](https://shopee.sg/Jinro-Soju-Green-Grape-360ml-i.319055539.9444526168)_ **TODAY**

~~I love CS~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

brendanng7 committed:
feat: renamed Duke to Soju

brendanng7 committed:
feat: added greet, exit functionalities as as well as adding horizontal line after each function

brendanng7 committed:
feat: added echoing functionality

brendanng7 committed:
feat: added adding of tasks and listing of tasks

brendanng7 committed:
feat: added new Task class

brendanng7 committed:
feat: added marking and unmarking of tasks in Soju

brendanng7 committed:
feat: added Todo, Deadline and Event classes

brendanng7 committed:
feat: added proper logic to echo

brendanng7 committed:
Add Text UI Testing

We must use Text UI Testing

brendanng7 committed:
Refactor add to list functions

brendanng7 committed:
Add exception handling for minimal cases

brendanng7 committed:
Fix on deadline command

brendanng7 committed:
Add deleting functionality

brendanng7 committed:
Add FileHandler class to handle file logic

brendanng7 committed:
Add storing of tasks in a .txt file

Restarting Soju will also reset the list of tasks.

We need to store the tasks for future reference.

Let's use a .txt file to store the information.

brendanng7 committed:
Merge pull request #1 from brendanng7/branch-Level-7

Branch level 7

brendanng7 committed:
Add LocalDate requirement for Deadline

brendanng7 committed:
Add chatbot date and time comprehension.

Strings were used to store date and times that had no common formatting.

We need a better way to streamline all dates and times.

Let's,
1. use LocalDate class to deal with dates/times
2. store the date and times in the storage file .txt as a string that the LocalDate/LocalDateTime parsers will recognise

brendanng7 committed:
Merge pull request #2 from brendanng7/branch-Level-8

Branch level 8

brendanng7 committed:
Add TaskList class to handle list of tasks

Tasks were stored in the Soju.class and made it cluttered.

There is a need for code clarity and seperation of concerns.

Lets create TaskList class that has the basic skeleton for manipulating tasks for now.

brendanng7 committed:
Ui.java: Add basic skeleton and functions

brendanng7 committed:
Use more OOP

All of the code logic was condensed into Soju.java. It was very difficult to make changes to the existing codebase.

We can seperate out the different code logic into seperate components.

Lets,
1. Create several relevant classes such as Ui, Storage, Parser and TaskList
2. Create commands packages to organise the file structure better
3. Perform regression testing to ensure no bugs

brendanng7 committed:
Refactor commands to not print line and only print lines in soju.java

brendanng7 committed:
Ui.java: Add Scanner field to handle user input

brendanng7 committed:
Merge pull request #3 from brendanng7/branch-A-MoreOOP

Branch a more oop

brendanng7 committed:
Merge branch 'add-gradle-support'

brendanng7 committed:
Add JUnit tests

brendanng7 committed:
build.gradle: Changes to configuration

brendanng7 committed:
Fix on build.gradle and line printing

brendanng7 committed:
Merge tag 'A-JUnit'

brendanng7 committed:
Add JavaDocs to classes in soju.tasks and soju

brendanng7 committed:
Add find functionality to TaskList

brendanng7 committed:
Merge branch 'branch-Level-9'

brendanng7 committed:
Refactor to follow coding standard

brendanng7 committed:
Add checkstyle configurations

All code styles were manually checked, leading to a lot of mistakes made.

There should be an automated way of checking the codestyle.

Lets,
1. Use Checkstyle to detect coding style violations
2. Refactor the codebase baseed on the mistakes pointed out by Checkstyle

brendanng7 committed:
Add javafx dependencies to build.gradle

brendanng7 committed:
Refactor code of test directory to fit coding standard

brendanng7 committed:
Change command::execute to return String response

Commands only used sout to print its response to the terminal.

We are using JavaFX now and will require to display the string in its labels.
Hence, we will keep the functionality of printing to the terminal but also return the response string.

Lets,
1) update the parent Command class to return a string in its execute method
2) update the rest of Command's children classes

brendanng7 committed:
Integrate JavaFX code into Soju

BunnyHoppp commented on src/main/java/soju/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
```
I like that you explicitly imported all the classes

BunnyHoppp commented on src/main/java/soju/Storage.java:
```diff
@@ -0,0 +1,143 @@
+                    break;
+                case "E":
+                    // Create a new Event task
+                    String[] eventTimes = parts[3].split(" - ");
```
I like the use of plural nouns for array.

BunnyHoppp commented on src/main/java/soju/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+     * @return A string representation of matching tasks, where each task is prefixed with its position number.
+     *         Returns an empty string if no tasks match the keyword.
+     */
+    public String findMatchingTasks(String keyword) {
```
I like how the method name explains the method and is not too long.

BunnyHoppp commented on src/main/java/soju/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+            }
+
+            // Extract the part after "deadline "
+            String[] parts = input.substring(9).split(" /by ", 2);
```
Perhaps you could have a clearer name instead of just parts?

jessica2828 replied:
Agree, i think more descriptive variable names would make your code easier to read!

BunnyHoppp commented on src/main/java/soju/tasks/Task.java:
```diff
@@ -0,0 +1,59 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
I like that the boolean variable sounds like a boolean

BunnyHoppp reviewed (commented):
I like the naming overall, very clean

jessica2828 commented on src/main/java/soju/commands/ByeCommand.java:
```diff
@@ -0,0 +1,18 @@
+/**
+ * ByeCommand handles commands starting with bye
+ */
+public class ByeCommand extends Command {
```
I like that you named the classes in your command package with -Command suffix, makes it easier to keep track

jessica2828 commented on src/main/java/soju/Ui.java:
```diff
@@ -0,0 +1,75 @@
+     * Reads a line of input from the user.
+     *
+     * @return The input entered by the user.
+     */
```
I like how you put detailed javadoc for all the methods across the files, really helpful in understanding your code

jessica2828 commented on src/main/java/soju/Storage.java:
```diff
@@ -0,0 +1,143 @@
+
+                switch (taskType) {
+                case "T":
+                    // Create a new Todo task
```
Good use of comments, helpful in understanding the code and they follow the coding style too 👍

jessica2828 commented on src/main/java/soju/commands/ByeCommand.java:
```diff
@@ -0,0 +1,18 @@
+ * ByeCommand handles commands starting with bye
+ */
+public class ByeCommand extends Command {
+    public ByeCommand(String input) {
+    }
```
Your ByeCommand constructor seems to not do anything, is there a reason why it was left empty?

jessica2828 commented:
Overall easy to read and well-explained code, and follows the coding standard. Well done!

brendanng7 committed:
Rename vague variable names

brendanng7 committed:
Change profile pictures of soju and user

brendanng7 committed:
Fix on delete command indexing and UI improvement

brendanng7 committed:
Add assert to TaskList

Tasklist methods such as unmark, mark and delete task work on a 1-based index.
Incorrect command inputs will lead to an indexing error.

Need to ensure assumption that user inputs correct list index.

Lets,
1) use Java assertions to ensure proper user inputs

brendanng7 committed:
Refactor variable naming to better code quality

brendanng7 committed:
Merge pull request #4 from brendanng7/branch-A-Assertions

Add assert to TaskList

brendanng7 committed:
Merge pull request #5 from brendanng7/branch-A-CodeQuality

Refactor variable naming to better code quality

brendanng7 committed:
Add .yml file for CI

brendanng7 committed:
Update .yml file

Project had no form of continuous integration.

CI helps to ensure that pull requests that are merged are automatically tested
instead of manually.

Lets,
1) Add CI using github workflows

brendanng7 committed:
Merge pull request #6 from brendanng7/branch-A-CI

Add .yml file for CI

brendanng7 committed:
Add anomaly detecting capabilities

Events could be added even if there were clashes with another added event.

User should be informed if the event is clashing with another event,
so that double booking will not happen.

Lets,
1) Modify Event to be able to check if it clashes with other Event objects
2) Update EventCommand's execute method

brendanng7 committed:
Remove unused import

brendanng7 committed:
Fix code quality issues

brendanng7 committed:
Add better GUI to Sojubot

All messages and commands had the same style.

We need to differentiate between user and soju commands.

Lets,
1) Add margin and spacings
2) Use fontstyles and colors to differentiate
3) Use anchors to size window correctly

brendanng7 committed:
Add personalised greeting message

brendanng7 committed:
Update AI.md

brendanng7 committed:
Add product snapshot and update user guide

brendanng7 committed:
Update README.md and app title

brendanng7 committed:
Update mainClassName in build.gradle

brendanng7 committed:
Fix fat jar file creation

brendanng7 committed:
EventCommand.java: Handle error of to date before for date

brendanng7 committed:
Add more test cases for parser and commands
